app/models/reports/schema_v2.rb

```diff
@@ -0,0 +1,230 @@
+    end
+
+    memoize def earliest_patient_recorded_at_period
+      earliest_patient_recorded_at.each_with_object({}) { |(slug, time), hsh| hsh[slug] = Period.new(value: time, type: @period_type) if time }
+    end
+
+    private def earliest_patient_data_query_v2(region)
+      FacilityState.for_region(region).where("cumulative_registrations > 0 OR cumulative_assigned_patients > 0")
+        .minimum(:month_date)
+    end
+
+    # Adjusted patient counts are the patient counts from three months ago (the adjusted period) that
+    # are the basis for control rates. These counts DO include lost to follow up.
+    memoize def adjusted_patients_with_ltfu
+      cumulative_assigned_patients.each_with_object({}) do |(entry, result), results|
+        values = periods.each_with_object(Hash.new(0)) { |period, region_result|
+          region_result[period] = result[period.adjusted_period]
+        }
+        results[entry] = values
+      end
+    end
+
+    # Adjusted patient counts are the patient counts from three months ago (the adjusted period) that
+    # are the basis for control rates. These counts DO NOT include lost to follow up.
+    memoize def adjusted_patients_without_ltfu
+      regions.each_with_object({}) { |region, result|
+        result[region.slug] = adjusted_patients_without_ltfu_query(region).slice(*periods.entries)
+      }
+    end
+
+    private def adjusted_patients_without_ltfu_query(region)
+      FacilityState.for_region(region)
+        .order(:month_date)
+        .group(:month_date)
+        .sum("patients_under_care::int")
+        .to_h(&period_hash)
+        .tap { |hsh| hsh.default = 0 }
+    end
+
+    alias_method :adjusted_patients, :adjusted_patients_without_ltfu
+
+    # Return the running total of cumulative assigned patient counts. Note that this *includes* LTFU.
+    memoize def cumulative_assigned_patients
+      regions.each_with_object({}) { |region, result|
+        result[region.slug] = cumulative_assigned_patients_query_v2(region)
+      }
+    end
+
+    # Returns cumulative assigned patients from facility_states - this includes LTFU
+    private def cumulative_assigned_patients_query_v2(region)
+      FacilityState.for_region(region)
+        .order(:month_date)
+        .group(:month_date)
+        .sum("cumulative_assigned_patients::int")
+        .to_h(&period_hash)
+        .tap { |hsh| hsh.default = 0 }
+    end
+
+    # Returns registration counts per region / period
+    memoize def monthly_registrations
+      regions.each_with_object({}) { |region, result|
+        result[region.slug] = registered_patients_query_v2(region).slice(*periods.entries)
+      }
+    end
+
+    def registered_patients_query_v2(region)
+      return {} if earliest_patient_recorded_at_period[region.slug].nil?
+      FacilityState.for_region(region)
+        .where("month_date >= ?", earliest_patient_recorded_at_period[region.slug].to_date)
+        .order(:month_date)
```

Comment:
Why is it necessary to order the months? Also, is the order preserved when doing a `to_h`? Same question for the other queries that do the same.